contracts/twap/CumulativeTwap.sol

```diff
@@ -22,10 +22,11 @@ contract CumulativeTwap is BlockContext {
     // STATE
     //
 
-    uint8 public currentObservationIndex;
+    uint16 public currentObservationIndex;
+    uint16 internal constant MAX_OBSERVATION = 1800;
     // let's use 15 mins and 1 hr twap as example
-    // if the price is updated every 15 secs, then we need 60 and 240 historical data for 15mins and 1hr twap
-    Observation[256] public observations;
+    // if the price is updated every 2 secs, 1hr twap Observation should have 60 / 2 * 60 = 1800 slots
+    Observation[MAX_OBSERVATION + 1] public observations;
```

Comment:
Why is here `MAX_OBSERVATION + 1`? Is this is ensure that 1hr twap will definitely succeed?

Reply:
Forgot to remove it at https://github.com/perpetual-protocol/perp-oracle-contract/pull/61/commits/09bc854d6867e3af60e1bb2fb66a5c6ad2263581. The original logic is directly `currentObservationIndex++;` to overflow it without the `%` calculation logic (L52). This means observations should be `Observation[MAX_OBSERVATION + 1]` (slots are 0-65535). To make the `currentObservationIndex` overflow from 65535. If we only assign like `Observation[MAX_OBSERVATION]`, the max index slot is 65534 which means the code will crash. 

However, after change to 1800, there's no way to natually overflow it. I need to remove that +1. Thanks for spoting it.